pkg/scheduler/algorithm/predicates/metadata.go

```diff
@@ -371,45 +370,46 @@ func getTPMapMatchingExistingAntiAffinity(pod *v1.Pod, nodeInfoMap map[string]*s
 		allNodeNames = append(allNodeNames, name)
 	}
 
-	var lock sync.Mutex
-	var firstError error
-
-	topologyMaps := newTopologyPairsMaps()
-
-	appendTopologyPairsMaps := func(toAppend *topologyPairsMaps) {
-		lock.Lock()
-		defer lock.Unlock()
-		topologyMaps.appendMaps(toAppend)
-	}
-	catchError := func(err error) {
-		lock.Lock()
-		defer lock.Unlock()
-		if firstError == nil {
-			firstError = err
-		}
-	}
+	allNodesCount := len(allNodeNames)
+	errCh := make(chan error, allNodesCount)
```

Comment:
Since we are only looking at a single error, you can make the buffer size equal to 1 and the send non-blocking as I describe below. This improves memory footprint.